File: abc087/abc087b.py
A = int(input())
B = int(input())
C = int(input())
X = int(input())

result = 0
for i in range(A + 1):
    for j in range(B + 1):
        for k in range(C + 1):
            if i * 500 + j * 100 + k * 50 == X:
                result += 1
print(result)

File: abc087/abc087b.go
package main

import (
	"fmt"
)

func main() {
	var a, b, c, x int
	fmt.Scan(&a, &b, &c, &x)
	result := 0
	for i := 0; i <= a; i++ {
		for j := 0; j <= b; j++ {
			for k := 0; k <= c; k++ {
				if x == 500*i+100*j+50*k {
					result++
				}
			}
		}
	}
	fmt.Println(result)
}
